### DISCUSSION TITLE
Session Based GraphQL Omit API

### DISCUSSION THREAD

**Discussion by noor-codes** [keystonejs/keystone#7315] [Feature requests]
## Session Based GraphQL Omit API

Currently the GraphQL API exposes anything that exists on the list but we can control what shouldn't be shown on the GraphQL API using `graphql.omit` option.

### This is what we have
```typescript
graphql: {
  omit: ['read', 'create', 'update']
}
```
### This is what we want
```typescript
graphql: {
  omit: ({ session }) => {
    if (!isAdmin({ session })) {
      return ['read', 'create', 'update']
    }
  }
}
```
If I have access to the `session` on omit option. I can determine who can be able to see things on the GraphQL API.
Right now, I can turn it off for everyone or no one.

**Comment by dcousens** [comment]
@noor-codes as is, the GraphQL schema is built statically by Keystone with `keystone build`,  how would you propose this works with sessions in respect to distributing that schema?
We had varying schemas in Keystone 5 and that ended up with confusion and misleading guarantees about what was possible.

**Reply by noor-codes** [reply]
Oh, that's complicated then. if the GraphQL API was generated on demand and we had access to the session. That would have been great. 

Thanks for the info @dcousens
